src/backend/joanie/core/api/admin.py

```diff
@@ -308,3 +308,35 @@ class ContractDefinitionViewSet(viewsets.ModelViewSet):
     permission_classes = [permissions.IsAdminUser & permissions.DjangoModelPermissions]
     serializer_class = serializers.AdminContractDefinitionSerializer
     queryset = models.ContractDefinition.objects.all()
+
+
+class OrderGroupViewSet(
+    mixins.CreateModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.ListModelMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    OrderGroup ViewSet
+    """
+
+    authentication_classes = [SessionAuthenticationWithAuthenticateHeader]
+    permission_classes = [permissions.IsAdminUser & permissions.DjangoModelPermissions]
+    serializer_class = serializers.AdminOrderGroupSerializer
+    queryset = models.OrderGroup.objects.all().select_related("product")
+    ordering = "created_on"
+
+    def get_queryset(self):
+        product_id = self.kwargs.get("product_id")
+        return super().get_queryset().filter(product__id=product_id)
+
+    def create(self, request, *args, **kwargs):
+        data = request.data
+        data["product"] = kwargs.get("product_id")
```

Comment:
Maybe I'm missing something but the variable `data` is not used at all

Reply:
`data` is changed and is used in `super().create()` by the serializer

Reply:
I confirm that if I delete these 2 lines, tests are still green.

Reply:
> `data` is changed and is used in `super().create()` by the serializer

You are right. This is too much magic for me but it works.

Reply:
In that case, too be more understandable, I suggest this

```suggestion
        request.data["product"] = kwargs.get("product_id")
```